### DISCUSSION TITLE
[@types/express] is incompatible with @types/node v16: No overload matches this call

### DISCUSSION THREAD

**Discussion by bennycode** [DefinitelyTyped/DefinitelyTyped#56683] [Issues with a @types package]
I have the following defined in my app:

```ts
app.use(bodyParser.json({limit: '50mb'}));
app.use(bodyParser.urlencoded({limit: '50mb', extended: true}));
``` 

And `app` is of type `express.Express` coming from:

```ts
import express from 'express';
```

The code works well in combination with `"@types/node": "15.6.1"` but it fails with `"@types/node": "16.11.2"` telling me:

> No overload matches this call.
>  The last overload gave the following error.
>    Argument of type 'NextHandleFunction' is not assignable to parameter of type 'PathParams'.
>      Type 'NextHandleFunction' is not assignable to type '(string | RegExp)[]'.
>
>13   app.use(bodyParser.urlencoded({limit: '50mb', extended: true}));
>             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>
>  node_modules/@types/express-serve-static-core/index.d.ts:109:5
>    109     <
>            ~
>    110         P = ParamsDictionary,
>        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>    ...
>    118         ...handlers: Array<RequestHandlerParams<P, ResBody, ReqBody, ReqQuery, Locals>>
>        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>    119     ): T;
>        ~~~~~~~~~
>    The last overload is declared here.

I am using `"@types/express": "4.17.13"`.

[a comment by an automated account is not shown]

**Comment by xylix** [comment]
I'm also having an issue with this but on my system it does not relate to node version. I created this issue https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/57382 to track the fact that it seems like the express types on the source tree don't even match the ones on NPM.

My current guess is that something somewhere pushed broken types into the @types/express package. But I don't really know how to validate if that is really to case.

**Reply by xylix** [reply]
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53584 This may be related
